lib/zero_push/compatibility.rb

```diff
@@ -1,20 +1,20 @@
 module ZeroPush
   module Compatibility
-    def notify(params)
-      params.symbolize_keys!
-      if (params[:info] || params[:data]).is_a?(String)
-        warn "[DEPRECATION] `info` or `data` should not be passed in as encoded strings."
+    def warn_on_deprecated_parameters(params)
+      value = (params[:info] || params['info'] || params[:data] || params['data'])
+      if value.is_a?(String)
+        warn "[DEPRECATION] `info` or `data` encoded as a string will not be supported in 2.5.1; Use a hash."
```

Comment:
Deprecation is a backwards incompatible change, this should require a major version release, not 2.5.1.